osdk/src/util.rs

```diff
@@ -290,3 +290,24 @@ impl Drop for DirGuard {
         std::env::set_current_dir(&self.0).unwrap();
     }
 }
+
+/// Attempts to create a hard link from `from` to `to`.
+/// If the hard link operation fails (e.g., due to crossing file systems),
+/// it falls back to performing a file copy.
+///
+/// # Arguments
+/// - `from`: The source file path.
+/// - `to`: The destination file path.
+///
+/// # Returns
+/// - `Ok(0)` if the hard link is successfully created (no data was copied).
+/// - `Ok(size)` where `size` is the number of bytes copied if the hard link failed and a copy was performed.
+/// - `Err(error)` if an error occurred during the copy operation.
+pub fn fast_copy<P: AsRef<Path>, Q: AsRef<Path>>(from: P, to: Q) -> Result<u64> {
```

Comment:
Might just rename it as `hard_link_or_copy` otherwise a user may assume that it is safe to modify the original file after copying.

Reply:
Sounds good! I will rename it.